Modern/wrapperPlugins/src/main/java/org/bonej/wrapperPlugins/wrapperUtils/UsageReporter.java

```diff
@@ -0,0 +1,294 @@
+/*
+BSD 2-Clause License
+Copyright (c) 2018, Michael Doube, Richard Domander, Alessandro Felder
+All rights reserved.
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+* Redistributions of source code must retain the above copyright notice, this
+  list of conditions and the following disclaimer.
+* Redistributions in binary form must reproduce the above copyright notice,
+  this list of conditions and the following disclaimer in the documentation
+  and/or other materials provided with the distribution.
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+package org.bonej.wrapperPlugins.wrapperUtils;
+
+import java.awt.Dimension;
+import java.awt.GraphicsConfiguration;
+import java.awt.GraphicsDevice;
+import java.awt.GraphicsEnvironment;
+import java.awt.Toolkit;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.charset.Charset;
+import java.util.Locale;
+import java.util.Random;
+
+import net.imagej.ImageJ;
+
+import org.scijava.Context;
+import org.scijava.command.ContextCommand;
+import org.scijava.log.LogService;
+import org.scijava.log.Logger;
+import org.scijava.plugin.Parameter;
+
+/**
+ * Prepare and send a report to be logged by 
+ * <a href="https://developers.google.com/analytics/resources/concepts/gaConceptsTrackingOverview">
+ * Google Analytics event tracking</a>
+ * <p>
+ * Should be called in a PlugIn's run() method as
+ * UsageReporter.reportEvent(this).send()
+ * </p>
+ * 
+ * @author Michael Doube
+ * @author Richard Domander
+ */
+public class UsageReporter extends ContextCommand {
+	
+	@Parameter
+	private static Context context;
+	
+	@Parameter
+	private static ImageJ imagej;
+	
+	@Parameter
+	private LogService logService;
+	
+	@Parameter
+	private Logger logger;
+	
+	public static final UsageReporter INSTANCE = new UsageReporter();
+	
+/**
+	 * BoneJ version FIXME: it is fragile to have the version hard-coded here.
+	 * Create a BoneJApp instead.
+	 */
+	private static final String BONEJ_VERSION = "2.0.0-experimental";
```

Comment:
> use the [`AppService`](http://javadoc.scijava.org/SciJava/org/scijava/app/AppService.html) for that ([`App.getArtifactId()`](http://javadoc.scijava.org/SciJava/org/scijava/app/App.html))

Sounds good, but how? Please point to an example, I'm having a great deal of trouble finding out how to get these new things to work

Reply:
I did a bit of investigating and found that you can use a `PluginService` for this:
```
// Any wrapper class from 'Modern' will do 
final String version = pluginService.getPlugin(ConnectivityWrapper.class).getVersion();
```
The version refers to the version of the [top level POM](https://github.com/bonej-org/BoneJ2/blob/master/pom.xml). This updates each time we release BoneJ, so it can be used to check which version someone is using. 

The next release will have version `7.0.0`. The numbering is a bit odd, because the POM started from `3.0.1` because of a miscommunication. I think it would be more confusing to try to fix the versioning by restarting from `1.0.0`, so I've accepted this oddity as "canon".

One slight problem in the future is that after modernising everything, we can get rid of the top-level POM, because BoneJ won't be split into `Legacy` and `Modern` any more. My best guess is that at this point `getVersion()` will return the version of the POM at the root of `Modern/`. To avoid confusion, maybe `BONEJ_VERSION` should be `getVersion() + "-experimental"` for now, and then changed when `Legacy` is removed.